SCClassLibrary/Common/Core/CondVar.sc

```diff
@@ -0,0 +1,116 @@
+CondVar {
+	var waitingThreads;
+
+	*new { ^super.newCopyArgs(Array.new, Array.new) }
+
+	wait { |predicate|
+		if(predicate.isNil) {
+			this.prWait
+		} {
+			// If the predicate is already true, we return immediately. If not,
+			// we block until the predicate is true after being signalled.
```

Comment:
For "bears of little brain" (in this case, including myself), it would be helpful to note in the comment that the condition may be signaled many times before the predicate becomes true -- hence, the loop is necessary. I honestly didn't understand that at first reading, and I expect others will struggle with it.